pkg/scheduler/cache/cache.go

```diff
@@ -1572,6 +1574,12 @@ func (sc *SchedulerCache) UpdateJobStatus(job *schedulingapi.JobInfo, updatePG b
 	return job, nil
 }
 
+func (sc *SchedulerCache) UpdateJobAnnotations(job *schedulingapi.JobInfo) {
```

Comment:
The cache update should put before API update call, or if API call failed we will miss set this filed in cache.